src/shim/hwctx.cpp

```diff
@@ -244,33 +244,14 @@ delete_ctx_on_device()
   fini_log_buf();
 }
 
-void
-hw_ctx::
-set_metadata(int num_cols, size_t size, uint64_t bo_paddr, uint8_t flag)
-{
-  m_metadata.magic_no = CERT_MAGIC_NO;
-  m_metadata.major = 0;
-  m_metadata.minor = 1;
-  m_metadata.cert_log_flag = flag;
-  m_metadata.num_cols = num_cols;
-  for (int i = 0; i < num_cols; i++) {
-    m_metadata.col_paddr[i] = bo_paddr + size * i + sizeof(m_metadata);
-    m_metadata.col_size[i] = size;
-  }
-}
-
 void
 hw_ctx::
 init_log_buf()
 {
-  size_t column_size = 1024;
-  auto log_buf_size = m_num_cols * column_size + sizeof(m_metadata);
+  auto log_buf_size = m_num_cols * 1024;
```

Comment:
this is incorrect size. Let me think how to handle this.

Reply:
Moving entire init_log_buf and related stuff to umq hwctx?